modules/network_synchronizer/networked_controller.cpp

```diff
@@ -0,0 +1,1585 @@
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+/**
+	@author AndreaCatania
+*/
+
+#include "networked_controller.h"
+
+#include "core/config/engine.h"
+#include "core/io/marshalls.h"
+#include "scene_synchronizer.h"
+#include <algorithm>
+
+#define METADATA_SIZE 1
+
+#define MAX_ADDITIONAL_TICK_SPEED 2.0
+
+// 2%
+#define TICK_SPEED_CHANGE_NOTIF_THRESHOLD 4
+
+StringName NetworkedController::sn_rpc_server_send_inputs;
+StringName NetworkedController::sn_rpc_send_tick_additional_speed;
+StringName NetworkedController::sn_rpc_doll_notify_sync_pause;
+StringName NetworkedController::sn_rpc_doll_send_epoch_batch;
+
+StringName NetworkedController::sn_controller_process;
+StringName NetworkedController::sn_count_input_size;
+StringName NetworkedController::sn_are_inputs_different;
+StringName NetworkedController::sn_collect_epoch_data;
+StringName NetworkedController::sn_collect_inputs;
+StringName NetworkedController::sn_setup_interpolator;
+StringName NetworkedController::sn_parse_epoch_data;
+StringName NetworkedController::sn_apply_epoch;
+
+void NetworkedController::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_player_input_storage_size", "size"), &NetworkedController::set_player_input_storage_size);
+	ClassDB::bind_method(D_METHOD("get_player_input_storage_size"), &NetworkedController::get_player_input_storage_size);
+
+	ClassDB::bind_method(D_METHOD("set_max_redundant_inputs", "max_redundand_inputs"), &NetworkedController::set_max_redundant_inputs);
```

Comment:
Perhaps “max_redundand_inputs” should have been “max_redundant_inputs”? :)

Reply:
Sure, thanks. Fixed the typo, will be merged into this branch with the update to the latest master soon.